Odata-docs/webapi-8/getting-started.md

```diff
@@ -0,0 +1,381 @@
+---
+title:  "Getting Started with ASP.NET Core OData 8"
+description: A getting started guide to creating and running an ASP.NET Core OData 8 application.
+date:   2022-11-11
+ms.date: 11/11/2022
+author: gathogojr
+ms.author: jogathog
+---
+
+# Getting Started with ASP.NET Core OData 8
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v8.md)]
+
+This tutorial shows how to create and run an ASP.NET Core OData 8 application
+
+You'll learn how to:  
+:white_check_mark: Create an ASP.NET Core application  
+:white_check_mark: Add Microsoft.AspNetCore.OData library package  
+:white_check_mark: Define the CLR types  
+:white_check_mark: Add an OData controller  
+:white_check_mark: Build the Edm model and configure the service  
+:white_check_mark: Run the OData service  
+:white_check_mark: Interact with the OData service from an API client  
+
+## Prerequisites
+- [Visual Studio 2022](https://visualstudio.microsoft.com/vs/#download) with the ASP.NET and web development workload - if target framework is [.NET Core 3.1](https://dotnet.microsoft.com/en-us/download/dotnet/3.1) or [.NET 6.0](https://dotnet.microsoft.com/en-us/download/dotnet/6.0), or
+- [Visual Studio 2019](https://visualstudio.microsoft.com/vs/older-downloads/) with the ASP.NET and web development workload - if target framework is [.NET Core 3.1](https://dotnet.microsoft.com/en-us/download/dotnet/3.1) 
+
+## Create an ASP.NET Core application
+In Visual Studio, create a new C# project from the **ASP.NET Core Empty** template. Name the project "Lab01".
+
+In the **Additional information** dialog, select either of **.NET Core 3.1 (Long Term Support)** or **.NET 6.0 (Long Term Support)** frameworks. Uncheck "Configure for HTTPS" checkbox. Click **Create**.
```

Comment:
The options available for .NET Core 3.1 and .NET 6.0 are so different that I don't think it's a good idea to combine them, otherwise someone could easily make a mistake and pick options that aren't compatible with OData (e.g. minimal APIs) or with the example here.

For example, if you're on .NET 6.0 and you select the  **ASP. NET Core Empty** template, you'll be presented with the option **Do not use top-level statements**. In my case, this is unchecked by default, so you do end up with top-level statements. That means in the `Program.cs` file doesn't contain a `Program` class or `Main` method.

![image](https://user-images.githubusercontent.com/8460169/201878719-1035e586-3e50-4a86-9984-ee72b735e48c.png)

If you check that option, you get a `Program` class and `Main` method.

In both cases, there is no `Startup` class. You use a `WebApplication.CreateBuilder()` to configure the web app and services.

Someone who's new to ASP.NET Core and OData might get lost if they follow along with this example that relies on `Startup.cs` class.

I think you should stick to a specific version and template and configuration for this sample. Or be detailed about the configurations for each template and version you want to include.

Reply:
@habbes They are actually not that different...

.NET Core 3.1 |  .NET 6.0
:------------:|:------------:
![Screenshot of creating aspnetcore project targeting netcore 3.1 framework](https://user-images.githubusercontent.com/28291332/202109642-f9e91e31-679a-465f-ab7e-18e19e785ba1.png)  |  ![Screenshot of creating aspnetcore project targeting net 6.0 framework](https://user-images.githubusercontent.com/28291332/202109682-278a89c9-2d0f-4446-9f33-91a216278073.png)

What I have done is add the above screenshots of creating the project using either frameworks in the doc side-by-side. I have also modified the section on how to configure the service to show how it should be done for both frameworks

Reply:
@gathogojr I see the options are similar for the Empty template, but the generated code is very different. In the .NET 6.0 version, you don't have a Startup.cs file, and the default is to use top-level statements.

Reply:
Yeah. For that reason I based the code snippet on the default option of using top-level statements